### PR TITLE
Block public access to all buckets

### PR DESCRIPTION
**Pull request by heronrs**
Blocks all public access to S3 rack-created buckets.

### CODE REVIEW

File: provider/aws/releases.go [hunk 1]
```diff
@@ -246,7 +246,7 @@ func (p *Provider) ReleasePromote(app, id string, opts structs.ReleasePromoteOpt
 			return err
 		}
 
-		ou, err := p.ObjectStore(app, "", bytes.NewReader(data), structs.ObjectStoreOptions{Public: options.Bool(true)})
+		ou, err := p.ObjectStore(app, "", bytes.NewReader(data), structs.ObjectStoreOptions{Presign: options.Bool(true)})
```
Comment (heronrs) [inline provider/aws/releases.go:249]: We are storing the new templates that CF will use for the deployment so we create a pre-signed URL - and CloudFormation will be able to download them.

File: provider/aws/formation/rack.json [hunk 2]
```diff
@@ -2845,7 +2845,7 @@
       "DependsOn": [ "Instances" ],
       "Properties": {
         "Cluster": { "Ref": "Cluster" },
-        "DeploymentConfiguration": { "MinimumHealthyPercent": "50", "MaximumPercent": "200" },
+        "DeploymentConfiguration": { "MinimumHealthyPercent": "50", "MaximumPercent": "200" },
```
Comment (ddoc) [inline provider/aws/formation/rack.json:2848]: added whitespace?

File: pkg/structs/object.go [hunk 3]
```diff
@@ -5,5 +5,6 @@ type Object struct {
 }
 
 type ObjectStoreOptions struct {
-	Public *bool `header:"Public"`
+	Public  *bool `header:"Public"`
+	Presign *bool
```
Comment (Twsouza) [inline pkg/structs/object.go:9]: Can `Public` field be deleted?

**Review by Twsouza — COMMENTED** [review]
Does switching from public to private interfere with current racks/apps?

**Comment by heronrs** [comment]
> Does switching from public to private interfere with current racks/apps?

Nope - I'll still run the e2e tests though.
The migration from public to non-public will occur at the rack level when the rack is updated and at the apps level when there's a new rollout.

**Review by Twsouza — APPROVED** [review]
:ship:

**Pull request closed without merging**